### ISSUE TITLE
gap in read not displayed as black bar

### ISSUE THREAD

**Issue by petermchale** [igvteam/igv.js#1058]
The attached screenshots indicate that the two reads that the popovers are "attached to" are in fact the same read. Yet there is no black bar connecting the different parts of the read. Is that a bug? If so, does it have something to do with the presence of the 1bp insertion immediately adjacent to the deletion (on its "right side")? 

Let me know if you want the bam. 

![left part of read](https://user-images.githubusercontent.com/19557137/73989119-edb3cd00-4901-11ea-8b6e-8ad918304c91.png)
![right part of read](https://user-images.githubusercontent.com/19557137/73989192-39ff0d00-4902-11ea-9795-3c1cebd19fd2.png)

**Comment by jrobinso** [comment]
Hi Peter.  This looks like a bug, and yes I suspect the insertion is involved as well.  The BAM would be helpful,  or a slice of the bam from this region.

**Comment by petermchale** [comment]
Mini bam at http://home.chpc.utah.edu/~u6018199/bams_igv_bug/
Reference is GRCh38.

**Comment by jrobinso** [comment]
@petermchale thanks for the bam, unfortunately I can't read.  Samtools error is ```[W::bam_hdr_read] EOF marker is absent. The input is probably truncated```.   Its only 245 bytes, does that seem correct?

**Comment by jrobinso** [comment]
Stepping through, the file is truncated after 245 bytes,  the first compressed block is at least 13,871 bytes according to the header.

**Comment by petermchale** [comment]
I downloaded the bam and ran `samtools quickcheck` (see attached screenshot). It seems to indicate that the bam contains an EOF: http://www.htslib.org/doc/samtools-quickcheck.html

The downloaded bam takes 4096 bytes on disk (see attached screenshot), so the fact that you report 245 bytes suggests that there was a problem downloading the file. Could you test this by comparing the output of `samtools view` you get with mine (see screenshot)? 

By the way, how did you reach the conclusion that "the first compressed block is at least 13,871 bytes according to the header"? I don't see any such info in the header (see screenshot). 

![Screen Shot 2020-02-10 at 11 48 20 AM](https://user-images.githubusercontent.com/19557137/74179773-57322500-4bfb-11ea-9831-55a102675eb9.png)

**Comment by jrobinso** [comment]
OK, my mistake, I was downloading directly from Chrome and it consistently stops at 245 bytes.  Using wget I got the whole file.

I got 13,871 from the bsize field in the compressed header block, but that is probably the uncompressed size.   In any event I think this bug is fixed, in master,  it will be ~2 weeks before we do a release.  Thanks for the report and test data.  Attached is a screenshot.

<img width="1012" alt="Screen Shot 2020-02-10 at 12 48 03 PM" src="https://user-images.githubusercontent.com/933148/74188756-f5c28400-4c03-11ea-84e7-d623feffe1dd.png">
